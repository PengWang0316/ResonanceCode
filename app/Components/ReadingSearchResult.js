import React from 'react';
// import PropTypes from 'prop-types';

import styles from '../styles/ReadingSearchResult.module.css';

const ReadingSearchResult = ({ element, nameResult, handleClickCallback }) => {
  const handleClick = _ => handleClickCallback(element._id, element.reading_name);
  return (
    <div role="button" tabIndex="-1" onClick={handleClick} className={`${styles.readingSearchItem}`}>{nameResult[1]}<span className={`${styles.matchKeyword}`}>{nameResult[2]}</span>{nameResult[3]}</div>
  );
};
// ReadingSearchResult.propTypes = {
//   element
// };
export default ReadingSearchResult;
