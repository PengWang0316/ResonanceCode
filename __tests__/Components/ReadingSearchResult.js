import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import ReadingSearchResult from '../../app/Components/ReadingSearchResult';

describe('ReadingSearchResult test', () => {
  const defaultProps = {
    element: { _id: 'id', reading_name: 'readingName' },
    nameResult: [1, 2, 3],
    handleClickCallback: jest.fn()
  };
  const getShallowComponent = (props = defaultProps) => shallow(<ReadingSearchResult {...props} />);

  test('ReadingSearchResult snapshot', () => expect(renderer.create(<ReadingSearchResult {...defaultProps} />).toJSON()).toMatchSnapshot());

  test('handleClick', () => {
    const component = getShallowComponent();
    component.find('div').simulate('click');
    expect(defaultProps.handleClickCallback)
      .toHaveBeenLastCalledWith(defaultProps.element._id, defaultProps.element.reading_name);
  });
});
