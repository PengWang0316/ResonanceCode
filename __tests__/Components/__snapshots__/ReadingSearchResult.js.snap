// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReadingSearchResult test ReadingSearchResult snapshot 1`] = `
<div
  className="readingSearchItem"
  onClick={[Function]}
  role="button"
  tabIndex="-1"
>
  2
  <span
    className="matchKeyword"
  >
    3
  </span>
</div>
`;
